src/services/limiter.ts

```diff
@@ -0,0 +1,82 @@
+import { JSONObject } from '@loopback/context'
+import { HttpErrors } from '@loopback/rest'
+import axios from 'axios'
+import { LimitError } from '../errors/types'
+import { blockHexToDecimal } from '../utils/block'
+
+import { WS_ONLY_METHODS } from '../utils/constants'
+
+const logger = require('../services/logger')
+
+export async function enforceEVMLimits(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  parsedRawData: Record<string, any>,
+  blockchainID: string,
+  logLimitBlocks: number,
+  altruists: JSONObject
+): Promise<void | Error> {
+  if (WS_ONLY_METHODS.includes(parsedRawData.method)) {
+    return new HttpErrors.BadRequest(
+      `We cannot serve ${parsedRawData.method} method over HTTPS. At the moment, we do not support WebSockets.`
+    )
+  } else if (parsedRawData.method === 'eth_getLogs') {
+    let toBlock: number
+    let fromBlock: number
+    let isToBlockHex = false
+    let isFromBlockHex = false
+    const altruistUrl = String(altruists[blockchainID])
```

Comment:
What if there's no `altruistUrl`? I think we should maybe throw an error here unless we intentionally don't use it if there's none?